src/Gaufrette/Adapter/AwsS3.php

```diff
@@ -9,31 +9,23 @@
 /**
  * Amazon S3 adapter using the AWS SDK for PHP v2.x.
  *
- * @author  Michael Dowling <mtdowling@gmail.com>
+ * @author Michael Dowling <mtdowling@gmail.com>
  */
 class AwsS3 implements Adapter, MetadataSupporter, ListKeysAware, SizeCalculator, MimeTypeProvider
 {
-    /** @var S3Client */
-    protected $service;
-    /** @var string */
-    protected $bucket;
-    /** @var array */
-    protected $options;
-    /** @var bool */
-    protected $bucketExists;
-    /** @var array */
-    protected $metadata = [];
-    /** @var bool */
-    protected $detectContentType;
-
-    /**
-     * @param S3Client $service
-     * @param string   $bucket
-     * @param array    $options
-     * @param bool     $detectContentType
-     */
-    public function __construct(S3Client $service, $bucket, array $options = [], $detectContentType = false)
-    {
+    protected S3Client $service;
+    protected string $bucket;
+    protected array $options;
+    protected bool $bucketExists;
```

Comment:
These should have a default value. I'm using your branch and getting these errors 

```
Uncaught Error: Typed property Gaufrette\Adapter\AwsS3::$bucketExists must not be accessed before initialization

"[\"/srv/api/vendor/knplabs/gaufrette/src/Gaufrette/Adapter/AwsS3.php:116\",\"/srv/api/vendor/knplabs/gaufrette/src/Gaufrette/Filesystem.php:103\",\"/srv/api/vendor/vich/uploader-bundle/src/Storage/GaufretteStorage.php:37\",\"/srv/api/vendor/vich/uploader-bundle/src/Storage/AbstractStorage.php:49\",\"/srv/api/vendor/vich/uploader-bundle/src/Handler/UploadHandler.php:50\",\"/srv/api/vendor/vich/uploader-bundle/src/EventListener/Doctrine/UploadListener.php:30\",\"/srv/api/vendor/symfony/doctrine-bridge/ContainerAwareEventManager.php:63\",\"/srv/api/vendor/doctrine/orm/lib/Doctrine/ORM/Event/ListenersInvoker.php:98\",\"/srv/api/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php:1033\",\"/srv/api/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php:1999\",\"/srv/api/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php:1956\",\"/srv/api/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php:665\",\"/srv/api/src/Services/DocumentService.php:54\",\"/srv/api/src/Services/DocumentUploadHelper.php:46\",\"/srv/api/src/Services/DocumentUploadHelper.php:34\",\"/srv/api/src/Validator/ClientInputInitializer.php:44\",\"/srv/api/vendor/symfony/validator/Validator/RecursiveContextualValidator.php:401\",\"/srv/api/vendor/symfony/validator/Validator/RecursiveContextualValidator.php:298\",\"/srv/api/vendor/symfony/validator/Validator/RecursiveContextualValidator.php:128\",\"/srv/api/vendor/symfony/validator/Validator/RecursiveValidator.php:82\",\"/srv/api/vendor/api-platform/core/src/Symfony/Validator/Validator.php:67\",\"/srv/api/src/DataTransformer/ClientInputDataTransformer.php:37\",\"/srv/api/vendor/api-platform/core/src/Serializer/AbstractItemNormalizer.php:305\",\"/srv/api/vendor/api-platform/core/src/JsonLd/Serializer/ItemNormalizer.php:166\",\"/srv/api/src/Serializer/ClientInputDenormalizer.php:59\",\"/srv/api/vendor/symfony/serializer/Serializer.php:234\",\"/srv/api/vendor/symfony/serializer/Serializer.php:152\",\"/srv/api/vendor/api-platform/core/src/Symfony/EventListener/DeserializeListener.php:137\",\"/srv/api/src/EventListener/Serializer/DeserializeListener.php:40\",\"/srv/api/vendor/symfony/event-dispatcher/EventDispatcher.php:260\",\"/srv/api/vendor/symfony/event-dispatcher/EventDispatcher.php:220\",\"/srv/api/vendor/symfony/event-dispatcher/EventDispatcher.php:56\",\"/srv/api/vendor/symfony/http-kernel/HttpKernel.php:157\",\"/srv/api/vendor/symfony/http-kernel/HttpKernel.php:76\",\"/srv/api/vendor/symfony/http-kernel/Kernel.php:197\",\"/srv/api/public/index.php:35\"]"
```

Reply:
Fixed by https://github.com/KnpLabs/Gaufrette/pull/716